src/allmydata/test/web/test_root.py

```diff
@@ -36,3 +45,363 @@ def test_missing(self):
         self.assertIdentical(res, ctx)
         self.assertEqual(ctx.slots["version"], "")
         self.assertEqual(ctx.slots["nickname"], "")
+
+
+class FakeUploader(object):
+    """
+    """
+    def get_helper_info(self):
+        return ("furl", False)
+
+
+class FakeHelper(object):
+    def get_stats(self):
+        return {
+            "chk_upload_helper.active_uploads": 0,
+        }
+
+
+class FakeStorageBroker(object):
+    def get_connected_servers(self):
+        return {}
+
+    def get_all_serverids(self):
+        return ()
+
+    def get_known_servers(self):
+        return ()
+
+
+class FakeMagicFolder(object):
+    def get_public_status(self):
+        return (True, ["this magic folder is alive"])
```

Comment:
Docstrings for all of these would be good.  Also, I am pretty sure there is already a fake for storage broker somewhere.  Or maybe it is possible to use a real storage broker with fake NativeStorageServers?  It would be great to use an existing (unverified) fake rather than creating a new (unverified) one.

Reply:
Yeah. I noted that for .. one of these. The main issue I see is that everything named `Fake*` that I've looked at isn't *really* a fake, but instead is "fake enough for this one test-module", approximately. :/ I'll see if we can use one of the other storage-broker fakes.

Reply:
deleted all these